tests/Catch2_Tests/comparators/ComparatorDisposerTest.cpp

```diff
@@ -0,0 +1,56 @@
+#include "catch2/catch.hpp"
+#include "ApprovalTests/comparators/ComparatorDisposer.h"
+#include "Approvals.h"
+#include <fstream>
+#include <iterator>
+#include <memory>
+
+using namespace ApprovalTests;
+
+namespace
+{
+    class TestComparator : public ApprovalComparator
+    {
+    public:
+        bool contentsAreEquivalent(std::string receivedPath,
+                                   std::string approvedPath) const override
+        {
+            std::ifstream receivedStream(receivedPath.c_str());
+            std::ifstream approvedStream(approvedPath.c_str());
+
+            using InputIter = std::istreambuf_iterator<char>;
+
+            auto receivedText =
+                std::string(InputIter{receivedStream}, InputIter{});
+            auto approvedText =
+                std::string(InputIter{approvedStream}, InputIter{});
+
+            return (receivedText == "foo\n") && (approvedText == "bar\n");
+        }
+    };
+}
+
+TEST_CASE("Type traits of ComparatorDisposer")
+{
+
+    REQUIRE_FALSE(std::is_copy_constructible<ComparatorDisposer>::value);
+    REQUIRE_FALSE(std::is_copy_assignable<ComparatorDisposer>::value);
+
+    REQUIRE(std::is_move_constructible<ComparatorDisposer>::value);
+    REQUIRE_FALSE(std::is_move_assignable<ComparatorDisposer>::value);
+}
+
+TEST_CASE("ComparatorDisposer object can be moved without disposing a "
+          "comparator twice")
+{
+    std::unique_ptr<ComparatorDisposer> disposer;
+
+    {
+        auto temporaryDisposer = FileApprover::registerComparatorForExtension(
+            ".txt", std::make_shared<TestComparator>());
+
+        disposer.reset(new ComparatorDisposer(std::move(temporaryDisposer)));
+    }
+
```

Comment:
There's a lot understand with this test... I'm only just learning about move semantics now, with the help of the new Josuttis book, and so it was really helpful to see this test. It took me a while to understand what was going on though.

As a suggestion, I think it could be simplified greatly by instead calling `ComparatorFactory::getComparatorForFileExtensionWithDot(".txt")` and checking that the return type is as expected... And removing the call to `Approvals::verify()` altogether.

I think the intention of the test would be clearer, and test-failures would be easier to understand...

Then the custom comparator for the test could be a trivial one, with no file-reading involved, or it could use one of the other provided comparators.

It would solve the other issues mentioned below, too...

Reply:
> It would solve the other issues mentioned below, too...

... mentioned above ...